python/GafferSceneUI/SetFilterUI.py

```diff
@@ -94,3 +100,162 @@
 	}
 
 )
+
+##########################################################################
+# NodeGadget drop handler
+##########################################################################
+
+GafferUI.Pointer.registerPointer( "sets", GafferUI.Pointer( "pointerSets.png", imath.V2i( 53, 14 ) ) )
+GafferUI.Pointer.registerPointer( "addSets", GafferUI.Pointer( "pointerAddSets.png", imath.V2i( 53, 14 ) ) )
+GafferUI.Pointer.registerPointer( "removeSets", GafferUI.Pointer( "pointerRemoveSets.png", imath.V2i( 53, 14 ) ) )
+GafferUI.Pointer.registerPointer( "replaceSets", GafferUI.Pointer( "pointerReplaceSets.png", imath.V2i( 53, 14 ) ) )
+
+__DropMode = enum.Enum( "__DropMode", [ "None_", "Add", "Remove", "Replace" ] )
+
+__originalDragPointer = None
+
+def __setsPlug( node ) :
+
+	for plug in node.children( Gaffer.Plug ) :
+		if Gaffer.Metadata.value( plug, "ui:scene:acceptsSetNames" ) or Gaffer.Metadata.value( plug, "ui:scene:acceptsSetExpression" ) :
+			return plug
+
```

Comment:
We need to check for settability and read-onliness somewhere - maybe in here? Currently you can drop onto a plug with an input and then get this :

```
ERROR : EventSignalCombiner : Cannot set value for plug "gui.scripts.ScriptNode1.SetFilter.setExpression" except during computation.
```

Reply:
Thanks for pointing that out, I hadn't considered read-onliness. I looked at adding that to `__setsPlug()`, but I think it might be better to check in `__dropMode()`. That's used by `__dragEnter()` to check if it can be dropped onto and I think the conditions are complicated enough that we'll have to add some conditions in `__dropMode()` anyways. And leaving `__setsPlug()` to return a plug accepting sets regardless of editability feels a little clearer to me.

The cases I have in mind are, given a `FilteredSceneProcessorNode`, potentially with a `SetFilter` input connection :

1. `FilteredSceneProcessorNode` editable, `SetFilter` not editable = no drop on either
2. `FilteredSceneProcessorNode` not editable, `SetFilter` not editable = no drop on either
3. `FilteredSceneProcessorNode` editable, `SetFilter` editable = drop on either
4. `FilteredSceneProcessorNode` not editable, `SetFilter` editable = drop on either? That's based on a strict interpretation of editability - the thing actually changing is the filter node and it's editable. Or do we make only the `SetFilter` droppable? Would any effect be unexpected if dropping onto a non-editable `FilteredSceneProcessorNode`?
5. `FilteredSceneProcessorNode` editable, no `SetFilter` connected = drop on `FilteredSceneProcessorNode`
6. `FilteredSceneProcessorNode` not editable, no `SetFilter` connected = no drop

I think that takes care of all the cases, any jump out that I'm missing?

Reply:
> I think it might be better to check in __dropMode(). 

Good call.

> I think that takes care of all the cases, any jump out that I'm missing?

That sounds good to me, and I agree that 4 should allow the drop, since the thing we want to edit in the end is editable. This also makes the implementation easier - we only need to check editability of the sets plug.